utils/mediaupload.js

```diff
@@ -31,37 +31,48 @@ export function mediaUpload( filesList, setAttributes, gallery = false ) {
 			setAttributes( media );
 		}
 
-		// Create upload payload
-		const data = new window.FormData();
-		data.append( 'file', mediaFile );
-
-		new wp.api.models.Media().save( null, {
-			data: data,
-			contentType: false,
-		} ).done( ( savedMedia ) => {
-			media.id = savedMedia.id;
-			media.url = savedMedia.source_url;
-			if ( gallery ) {
-				setAttributes( { images: [
-					...gallerySet.slice( 0, idx ),
-					media,
-					...gallerySet.slice( idx + 1 ),
-				] } );
-			} else {
-				setAttributes( media );
-			}
-		} ).fail( () => {
-			// Reset to empty on failure.
-			// TODO: Better failure messaging
-			media.url = null;
-			if ( gallery ) {
-				setAttributes( { images: [
-					...gallerySet.slice( 0, idx ),
-					...gallerySet.slice( idx + 1 ),
-				] } );
-			} else {
-				setAttributes( media );
+		return createMediaFromFile( mediaFile ).then(
+			( savedMedia ) => {
+				media.id = savedMedia.id;
+				media.url = savedMedia.source_url;
+				if ( gallery ) {
+					setAttributes( { images: [
```

Comment:
I know this existed previously, but it seems odd to me that we need to handle the `setAttributes` logic here, especially because it forces us to fork the logic depending on the block type. Shouldn't this function just return the promise that resolves as the `savedMedia` object, encapsulating only the steps necessary to create and send the media payload?